src/directory/mmap_directory.rs

```diff
@@ -479,6 +479,7 @@ impl Directory for MmapDirectory {
         let file: File = OpenOptions::new()
             .write(true)
             .create(true) //< if the file does not exist yet, create it.
+            .truncate(false)
```

Comment:
Wouldn't this make more sense?

```suggestion
            .create_new(true) //< if the file already exists, fail
```

Reply:
This would disable blocking locking. I'm also not sure if we delete the lock

Reply:
Ah I totally missed the context. I thought it was for regular writes!